src/sql/workbench/services/objectExplorer/browser/treeSelectionHandler.ts

```diff
@@ -93,44 +94,54 @@ export class TreeSelectionHandler {
 	 *
 	 * @param connectionCompleteCallback A function that gets called after a connection is established due to the selection, if needed
 	 */
-	private handleTreeItemSelected(connectionManagementService: IConnectionManagementService, objectExplorerService: IObjectExplorerService, isDoubleClick: boolean, isKeyboard: boolean, selection: any[], tree: ITree, connectionCompleteCallback: () => void): void {
-		let connectionProfile: ConnectionProfile = undefined;
-		let options: IConnectionCompletionOptions = {
-			params: undefined,
-			saveTheConnection: true,
-			showConnectionDialogOnError: true,
-			showFirewallRuleOnError: true,
-			showDashboard: isDoubleClick // only show the dashboard if the action is double click
-		};
-		if (selection && selection.length > 0 && (selection[0] instanceof ConnectionProfile)) {
-			connectionProfile = <ConnectionProfile>selection[0];
-
-			if (connectionProfile) {
-				this.onTreeActionStateChange(true);
-
-				TreeUpdateUtils.connectAndCreateOeSession(connectionProfile, options, connectionManagementService, objectExplorerService, tree).then(sessionCreated => {
-					if (!sessionCreated) {
-						this.onTreeActionStateChange(false);
-					}
-					if (connectionCompleteCallback) {
-						connectionCompleteCallback();
-					}
-				}, error => {
-					this.onTreeActionStateChange(false);
-				});
+	private handleTreeItemSelected(connectionManagementService: IConnectionManagementService, objectExplorerService: IObjectExplorerService, isDoubleClick: boolean, isKeyboard: boolean, selection: any[], tree: AsyncServerTree | ITree, connectionCompleteCallback: () => void): void {
+		if (tree instanceof AsyncServerTree) {
+			let connectionProfile: ConnectionProfile = undefined;
+			if (selection && selection.length > 0 && (selection[0] instanceof ConnectionProfile)) {
+				connectionProfile = <ConnectionProfile>selection[0];
```

Comment:
The new recommended way to cast is using `as` so something like `selection[0] as ConnectionProfile`

Reply:
True, although in this case I can also just rid of the whole casting thing anyways since the instanceof handles that for us.